### DISCUSSION TITLE
@lit/localize: Add a way to remove the 'desc' from msg() options from a production build (to save kb's)

### DISCUSSION THREAD

**Discussion by tomvlk** [lit/lit#4849] [Ideas]
Hi there, We are going to heavily use the lit/localize package for our entire PWA. And for many strings/words we want to add some context with the `desc` option in the `msg()` function. What I noticed is that this also ends up in the production (which makes sense because it's JS code).

We are using esbuild, and I'm wondering if we can come up with some smart way of removing it from a production build in order to save lot's of KB's and also potentially some context that you don't want to end up in the end-users device.

**Comment by Lalem001** [comment]
I wrote a Vite plugin some time ago that does this for my company. I will not have access to my computer till next week because of the holiday. I can certainly make a gist or similar when I get back.

**Comment by Lalem001** [comment]
@tomvlk Sorry for the delay, I have been prepping a npm releasable version of my plugin while I await approval from legal at my company. But I also realize that my plugin for Vite (rollup) would not fit your needs for an `esbuild` specific tool. Perhaps you could use my plugin with rollup as a post processing step?

**Reply by tomvlk** [reply]
No worries @Lalem001 , let me know. I might consider creating something similar for `esbuild` and publish potentially. Keep me in the loop if you are allowed to publish it.

**Comment by Lalem001** [comment]
@tomvlk I was given permission to share via gist, please see it at https://gist.github.com/Lalem001/acea6c33c89c20e0d1212328e4918d2f

I included comments next to the imports denoting the dependency versions I was testing against.
